### ISSUE TITLE
False alarm from new W4002

### ISSUE THREAD

**Issue by schmiddy** [aws-cloudformation/cfn-python-lint#1625]
*cfn-lint version: 0.34.0*

[Here](https://gist.github.com/schmiddy/44a779032a930995d22ee2722a18f163) is an example template which causes a false alarm like this:

```
$ cfn-lint /tmp/example.yml 
W4002 As the resource "metadata" section contains reference to a "NoEcho" parameter DBUser, CloudFormation will display the parameter value in plaintext
/tmp/example.yml:21:7

W4002 As the resource "metadata" section contains reference to a "NoEcho" parameter DBPass, CloudFormation will display the parameter value in plaintext
/tmp/example.yml:21:7
```

The problem seems to be that the rule is looking for any mention of the parameter name, even as a text description that is not actually referencing the parameter.
